docs/regmap/adi_regmap_i3c_controller.txt

```diff
@@ -0,0 +1,609 @@
+ENDREG
+
+FIELD
+[31:0] 0x00000000
+IRQ_SOURCE
+RO
+Pending IRQs without mask.
+ENDFIELD
+
+############################################################################################
+############################################################################################
+
+REG
+0x30
+CMD_FIFO_ROOM
+ENDREG
+
+FIELD
+[31:0]
+CMD_FIFO_ROOM
+RO
+Number of free entries in the CMD FIFO.
+ENDFIELD
+
+############################################################################################
+############################################################################################
+
+REG
+0x31
+CMDR_FIFO_LEVEL
+ENDREG
+
+FIELD
+[31:0] 0x00000000
+CMDR_FIFO_LEVEL
+RO
+Number of valid entries in the CMDR FIFO.
+ENDFIELD
+
+############################################################################################
+############################################################################################
+
+REG
+0x32
+SDO_FIFO_ROOM
+ENDREG
+
+FIELD
+[31:0]
+SDO_FIFO_ROOM
+RO
+Number of free entries in the SDO FIFO.
+ENDFIELD
+
+############################################################################################
+############################################################################################
+
+REG
+0x33
+SDI_FIFO_LEVEL
+ENDREG
+
+FIELD
+[31:0] 0x00000000
+SDI_FIFO_LEVEL
+RO
+Number of valid entries in the serial-data-in FIFO.
+ENDFIELD
+
+############################################################################################
+############################################################################################
+
+REG
+0x34
+IBI_FIFO_LEVEL
+ENDREG
+
+FIELD
+[31:0] 0x00000000
+IBI_FIFO_LEVEL
+RO
+Number of valid entries in the in-bus-interrupt FIFO.
+ENDFIELD
+
+############################################################################################
+############################################################################################
+
+REG
+0x35
+CMD_FIFO
+Command FIFO register. Writing to this register inserts an entry into the CMD FIFO.
+Writing to this register when the CMD FIFO is full has no effect and the written entry
+is discarded. Reading from this register always returns 0x00000000.
+The Software is responsive for a valid sequence of commands.
+If the peripheral does not ACK when required during a command, the procedure exits and the
+next command in the FIFO is interpreted.
+See :ref:`i3c_controller instruction-format` for the structure of the command.
+ENDREG
+
+FIELD
+[31:0]
+CMD_FIFO
+WO
+The command to the I3C controller to execute.
+ENDFIELD
+
+############################################################################################
+############################################################################################
+
+REG
+0x36
+CMDR_FIFO
+CMDR FIFO register. Reading from this register removes the first entry from the CMDR FIFO.
+Reading this register when the CMDR FIFO is empty will return undefined data.
+Writing to it has no effect.
+ENDREG
+
+FIELD
+[31:0] 0x??
```

Comment:
question marks

Reply:
This one is our current discussion: Readable FIFOs